benchmark/Streamly/Benchmark/Data/Parser.hs

```diff
@@ -262,6 +266,7 @@ o_1_space_serial value =
     [ benchIOSink value "takeEQ" $ takeEQ value
     , benchIOSink value "takeWhile" $ takeWhile value
     , benchIOSink value "drainWhile" $ drainWhile value
+    , benchIOSink value "sliceBeginWith" $ sliceBeginWith value
```

Comment:
Are the benchmarks similar to `takeWhile`?

Reply:
> @specdrake Please address the comments.

Done, sorry for the delay.

Reply:
```
Data.Parser/o-1-space/sliceBeginWith     2238.86
Data.Parser/o-1-space/takeWhile            53.90
```
```
Data.Parser/o-1-space/sliceBeginWith     7307320.00
Data.Parser/o-1-space/takeWhile                0.00
```
```
Data.Parser/o-1-space/sliceBeginWith        2312.00
Data.Parser/o-1-space/takeWhile                0.00
```
But they are similar to those of `parseMany` and `some` and others.

Reply:
```
Data.Parser/o-1-space/sliceBeginWith     66.66
Data.Parser/o-1-space/takeWhile             51.36
```
```
Data.Parser/o-1-space/sliceBeginWith     0.00
Data.Parser/o-1-space/takeWhile            0.00
```
```
Data.Parser/o-1-space/sliceBeginWith     0.00
Data.Parser/o-1-space/takeWhile            0.00
```
```
Data.Parser/o-1-space/sliceBeginWith        66.79
Data.Parser/o-1-space/takeWhile                51.26
```
```
Data.Parser/o-1-space/sliceBeginWith     0.00
Data.Parser/o-1-space/takeWhile            0.00
```
```
Data.Parser/o-1-space/sliceBeginWith     0.00
Data.Parser/o-1-space/takeWhile            0.00
```